### PR TITLE
Remove GetRemotePackageVersion and Add Get<registry>Version in protos 

### PR DESCRIPTION
**Pull request by joshcarp**
Added directly from https://github.com/bufbuild/buf/pull/2182?notification_referrer_id=NT_kwDOAfGGmrM2NzU2MzU3MDk5OjMyNjA1ODUw#pullrequestreview-1491830479 but I don't want to merge that because it's not in main in core yet and it can't be added to core yet because it isn't in main on buf yet so I'm making this PR so that I can implement it in core and then rebase buf

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 1]
```diff
@@ -33,10 +33,24 @@ service ResolveService {
   rpc GetModulePins(GetModulePinsRequest) returns (GetModulePinsResponse) {
     option idempotency_level = NO_SIDE_EFFECTS;
   }
-
-  // GetRemotePackageVersion resolves the given plugin and module references to a version.
+  // GetGoVersion resolves the given plugin and module references to a version.
+  // The format of the version is determined by the plugin type.
```
Comment (stefanvanburen) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:37]:
```suggestion
```

I think you can nix the second line for each of these, as the format of the version is fixed per-registry.

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 2]
```diff
@@ -33,10 +33,24 @@ service ResolveService {
...
+  rpc GetGoVersion(GetGoVersionRequest) returns (GetGoVersionResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+  // GetSwiftVersion resolves the given plugin and module references to a version.
   // The format of the version is determined by the plugin type.
```
Comment (stefanvanburen) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:42]:
```suggestion
```

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 3]
```diff
@@ -33,10 +33,24 @@ service ResolveService {
...
-  rpc GetRemotePackageVersion(GetRemotePackageVersionRequest) returns (GetRemotePackageVersionResponse) {
+  rpc GetSwiftVersion(GetSwiftVersionRequest) returns (GetSwiftVersionResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+  // GetMavenVersion resolves the given plugin and module references to a version.
+  // The format of the version is determined by the plugin type.
```
Comment (stefanvanburen) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:47]:
```suggestion
```

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 4]
```diff
@@ -33,10 +33,24 @@ service ResolveService {
...
+  rpc GetMavenVersion(GetMavenVersionRequest) returns (GetMavenVersionResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+  // GetNPMVersion resolves the given plugin and module references to a version.
+  // The format of the version is determined by the plugin type.
```
Comment (stefanvanburen) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:52]:
```suggestion
```

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 5]
```diff
@@ -114,13 +128,54 @@ message GetLocalModulePinsResponse {
...
+
+message GetMavenVersionRequest {
   // The plugin reference to resolve.
   GetRemotePackageVersionPlugin plugin_reference = 1;
   // The module reference to resolve.
   LocalModuleReference module_reference = 2;
 }
 
+message GetMavenVersionResponse {
+  // version is the resolved version to be used with the go module proxy.
```
Comment (stefanvanburen) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:151]:
```suggestion
  // version is the resolved version to be used with the maven repository.
```

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 6]
```diff
@@ -114,13 +128,54 @@ message GetLocalModulePinsResponse {
...
+
+message GetNPMVersionRequest {
+  // The plugin reference to resolve.
+  GetRemotePackageVersionPlugin plugin_reference = 1;
+  // The module reference to resolve.
+  LocalModuleReference module_reference = 2;
+}
+
+message GetNPMVersionResponse {
+  // version is the resolved version to be used with the go module proxy.
```
Comment (stefanvanburen) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:163]:
```suggestion
  // version is the resolved version to be used with the npm registry.
```

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 7]
```diff
@@ -114,13 +128,54 @@ message GetLocalModulePinsResponse {
...
+
+message GetSwiftVersionRequest {
+  // The plugin reference to resolve.
+  GetRemotePackageVersionPlugin plugin_reference = 1;
+  // The module reference to resolve.
+  LocalModuleReference module_reference = 2;
+}
+
+message GetSwiftVersionResponse {
+  // version is the resolved version to be used with the go module proxy.
```
Comment (stefanvanburen) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:175]:
```suggestion
  // version is the resolved version to be used with the swift repository.
```

**Review by stefanvanburen — COMMENTED** [review]
looks fine, just needs updates to comments

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 8]
```diff
@@ -114,13 +124,54 @@ message GetLocalModulePinsResponse {
...
+
+message GetMavenVersionRequest {
+  // The plugin reference to resolve.
+  GetRemotePackageVersionPlugin plugin_reference = 1;
+  // The module reference to resolve.
+  LocalModuleReference module_reference = 2;
+}
+
+message GetMavenVersionResponse {
+  // version is the resolved version to be used with the maven registry.
```
Comment (stefanvanburen) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:147]:
```suggestion
  // version is the resolved version to be used with the maven repository.
```

Comment (joshcarp) [reply proto/buf/alpha/registry/v1alpha1/resolve.proto:147]: We use the term `registry` everywhere don't we?

Comment (joshcarp) [reply proto/buf/alpha/registry/v1alpha1/resolve.proto:147]: nvm apparently it's different for maven

**Review by stefanvanburen — APPROVED** [review]

**Pull request merged**